gammapy/catalog/hawc.py

```diff
@@ -0,0 +1,134 @@
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        ss = '\n*** Basic info ***\n\n'
+        ss += 'Catalog row index (zero-based) : {}\n'.format(d['catalog_row_index'])
+        ss += '{:<15s} : {}\n'.format('HWC2_FHL_name:', d['source_name'])
+        return ss
+
+    def _info_position(self):
+        """Print position info."""
+        d = self.data
+        ss = '\n*** Position info ***\n\n'
+
+        # TODO: Add here the SIMBAD position???
+
+        ss += 'Measurement:\n'
+        ss += '{:20s} : {:.3f}\n'.format('RA', d['ra'])
+        ss += '{:20s} : {:.3f}\n'.format('DEC', d['dec'])
+        ss += '{:20s} : {:.3f}\n'.format('GLON', d['glon'])
+        ss += '{:20s} : {:.3f}\n'.format('GLAT', d['glat'])
+        ss += '{:20s} : {:.3f}\n'.format('Position error', d['pos_err'])
+
+        return ss
+
+    def _info_spectra(self):
+        """Print spectral info."""
+        d = self.data
+        ss = '\n*** Spectral info ***\n\n'
+
+        ss += 'Spectrum 1:\n'
+        ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7TeV', d['spec0_dnde'].value, \
+            d['spec0_dnde_err'].value, d['spec0_dnde'].unit)
+        ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('index', d['spec0_index'], d['spec0_index_err'])
+        ss += '{:20s} : {:1}\n\n'.format('Test radius', d['spec0_radius'])
+
+        if(np.isnan(d['spec1_dnde'])):
+            ss += 'No second spectrum available for this source'
+        else:
+            ss += 'Spectrum 2:\n'
+            ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7TeV', d['spec1_dnde'].value, \
+                d['spec1_dnde_err'].value, d['spec1_dnde'].unit)
+            ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('index', d['spec1_index'], d['spec1_index_err'])
+            ss += '{:20s} : {:1}'.format('Test radius', d['spec1_radius'])
+        return ss
+
+    # TODO: Add other information to def ___str___(self, info)???
+
+
+class SourceCatalog2HWC(SourceCatalog):
+    """ HAWC 2FHL catalog
+
+    One source is represented by `~gammapy.catalog.SourceCatalogObjectGammaCat`
+
+    See: TODO: Add here the reference_id?
+
+    Parameters: TODO: list the self.filename variable if decided to put it as variable
+
+    Examples
+    --------
+    TODO: Add examples
+    """
+
+    name = '2hwc' #TODO: Different name, e.g. '2hwc fhl' (if changed, needs to be changed in test_hawc.py as well)
+    description = ' ' # TODO: Add description
+    source_object_class = SourceCatalogObject2HWC  #TODO: Does not work, don't know why
+
+    def __init__(self, filename='$GAMMAPY_EXTRA/datasets/catalogs/2HWC.ecsv'):
+        filename = str(make_path(filename))
+
+        #TODO: initialise self.filename? In gammacat it is, but Hess and fermi it is not
+
+        table = Table.read(filename, format='ascii.ecsv', delimiter=' ')
+
+        source_name_key = 'source_name' #TODO: Call it e.g. 2HWC_FHL_name?
```

Comment:
Keep column names as-is in the ECSV file. Remove TODO here.

Reply:
Fixed